### ISSUE TITLE
mapControls props not kicking in?

### ISSUE THREAD

**Issue by Jauny** [uber/react-map-gl#384]
I'm trying to use map control to react to events such as 'mouseup'. Following the example, it doesn't seem like the `handle` function is called when my event happens;
```js
//map-controls.js
import {experimental} from 'react-map-gl';

export default class MyMapControls extends experimental.MapControls {
  // override the default handler in MapControls
  handle(event) {
    if (event.type === 'click') {
      console.log('hi!')
    }
    return super.handleEvent(event);
  }
}
```

```js
// map.js
import React, {Component} from 'react';
import ReactMapGL from 'react-map-gl';
import MyMapControls from './map-controls';

export default class Map extends Component {
  constructor(props) {
    super(props);
    this.state = {
      latitude: 37.7577,
      longitude: -122.4376
    };
  }

  render() {
    return (
      <ReactMapGL
        width={400}
        height={400}
        latitude={this.state.latitude}
        longitude={this.state.longitude}
        zoom={8}
        mapControls={new MyMapControls()}
      />
    );
  }
}
```

The documentation mentions that `events`, an array of handled events has to be passed but I can't figure out how, and the example doesn't pass any events to the props.

This code doesn't produce any error, but nothing happens when I click the map (tried with other events such as 'mousedown', 'mouseup').

**Comment by Pessimistress** [comment]
@Jauny The handler should be `handleEvent` instead of `handle`. There's an error the docs and the fix has not been published. :(

**Comment by Jauny** [comment]
@Pessimistress yeah unfortunately I tried that too. Here is code while I follow exactly the example, using `handleEvent` instead. Still nothing:

```js
/// map-controls.js
import {experimental} from 'react-map-gl';

export default class MyMapControls extends experimental.MapControls {
  constructor(props) {
    super(props);
    this.events = ['click'];
  }

  // override the default handler in MapControls
  handleEvent(event) {
    if (event.type === 'click') {
      console.log('hi!')
    }
    return super.handleEvent(event);
  }
}
```

```js
/// map.js
import React, {Component} from 'react';
import ReactMapGL from 'react-map-gl';
import MyMapControls from './map-controls';

export default class Map extends Component {
  constructor(props) {
    super(props);
    this.state = {
      latitude: 37.7577,
      longitude: -122.4376
    };
  }

  render() {
    return (
      <ReactMapGL
        width={400}
        height={400}
        latitude={this.state.latitude}
        longitude={this.state.longitude}
        zoom={8}
        mapControls={new MyMapControls()}
      />
    );
  }
}

**Comment by Pessimistress** [comment]
Which version are you using?

**Comment by Pessimistress** [comment]
Fixed in 3.1.1
